### PR TITLE
feat(comp): extend Box to handle __default variants

### PR DESCRIPTION
**Pull request by ti10le**
#1486

### CODE REVIEW

File: packages/system/src/Box.tsx [hunk 1]
```diff
@@ -69,8 +69,19 @@ export type BoxProps = PolymorphicPropsWithRef<BoxOwnProps, 'div'>;
 const isNotEmpty = (val: any) =>
   !(val && Object.keys(val).length === 0 && val.constructor === Object);
 
-const ensureArray = <T extends any>(val?: T | T[]) =>
-  Array.isArray(val) ? val : [val];
+/**
+ * Ensure that variant is an array and remove dot at the end of the string
+ * to set __default in theme without giving a default variant to the component
+ */
+const ensureCorrectVariant = <T extends string>(variant?: T | T[]) => {
```
Comment (sebald) [inline packages/system/src/Box.tsx:76]: Can we make this function more simple to read please? :D

**Review by sebald — CHANGES_REQUESTED** [review]

File: packages/system/src/Box.tsx (hunk 1 above)
Comment (ti10le) [reply packages/system/src/Box.tsx]: Better? Or how can I make it easier?

File: packages/system/src/Box.tsx [hunk 2]
```diff
@@ -72,6 +72,19 @@ const isNotEmpty = (val: any) =>
 const ensureArray = <T extends any>(val?: T | T[]) =>
   Array.isArray(val) ? val : [val];
 
+/**
+ * Ensure that variant is an array and remove dot at the end of the string
+ * to set __default in theme without giving a default variant to the component
+ */
+const ensureCorrectVariant = (variant?: any | any[]) => {
```
Comment (sebald) [inline packages/system/src/Box.tsx:79]: can only be string or string array, right?

File: packages/system/src/Box.tsx [hunk 3]
```diff
@@ -72,6 +72,19 @@ const isNotEmpty = (val: any) =>
...
+  if (variant) {
+    return ensureArray(variant).map(v => {
+      return v.replace(/\.$/, '');
```
Comment (sebald) [inline packages/system/src/Box.tsx:82]: unnecessary return, use shorthand? :)

File: packages/system/src/Box.tsx [hunk 4]
```diff
@@ -72,6 +72,19 @@ const isNotEmpty = (val: any) =>
...
+    });
+  }
+  return [variant];
```
Comment (sebald) [inline packages/system/src/Box.tsx:85]: This will always return [undefined]. Why do you need this here :)

File: packages/system/src/Box.tsx (hunk 2 above)
Comment (ti10le) [reply packages/system/src/Box.tsx:79]: or undefined

**Review by sebald — APPROVED** [review]

**Pull request merged**